core/src/main/java/ai/timefold/solver/core/impl/heuristic/move/AbstractMetricMove.java

```diff
@@ -0,0 +1,22 @@
+package ai.timefold.solver.core.impl.heuristic.move;
+
+import ai.timefold.solver.core.api.domain.solution.PlanningSolution;
+
+/**
+ * Abstract superclass for {@link Move} which includes the ability to enable or disable the move metric collection.
+ *
+ * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
+ */
+public abstract class AbstractMetricMove<Solution_> implements Move<Solution_> {
```

Comment:
Please think of a different way of doing this. `Move` is something that people extend; we don't want them to think about metrics as they do so. Moves should not have to understand if they're being measured or not.